meta/saisanitycheck.c

```diff
@@ -849,14 +849,13 @@ void check_attr_conditions(
             /* condition needs to be changed to validonly since flags should be CREATE_AND_SET */
 
             if (md->objecttype != SAI_OBJECT_TYPE_MIRROR_SESSION &&
-                    md->objecttype != SAI_OBJECT_TYPE_HOSTIF_TRAP &&
-                    md->objecttype != SAI_OBJECT_TYPE_HOSTIF_USER_DEFINED_TRAP)
+                    md->objecttype != SAI_OBJECT_TYPE_HOSTIF_TRAP)
             {
                 META_ASSERT_FAIL(md, "marked as conditional on non mirror session /hostif_trap/hostif_user_defined_trap, but invalid creation flags: 0x%u", md->flags);
             }
 
             break;
-
+        case SAI_ATTR_FLAGS_MANDATORY_ON_CREATE | SAI_ATTR_FLAGS_CREATE_ONLY | SAI_ATTR_FLAGS_KEY:
```

Comment:
Based on discussion Guohan and Matty had, and additional inputs, it was decided to unify trap and user defined trap to one object
Perhaps we should have more notions of keys ? Primary key (unique non null, meaning mandatory) and unique key (conditional, allow nulls, unique when value is used). In this case, we will use the unique key, vs in other cases we will use primary key when the attribute is mandatory